crates/dojo-core/src/database/index.cairo

```diff
@@ -12,16 +12,22 @@ struct WhereCondition {
     value: felt252,
 }
 
-fn create(address_domain: u32, index: felt252, id: felt252) {
-    if exists(address_domain, index, id) {
+fn create(address_domain: u32, table: felt252, id: felt252, value: felt252) {
+    if exists(address_domain, table, id) {
         return ();
     }
 
-    let index_len_key = build_index_len_key(index);
+    let index_len_key = build_index_len_key(table, value);
     let index_len = storage::get(address_domain, index_len_key);
-    storage::set(address_domain, build_index_item_key(index, id), index_len + 1);
+
+    if value == 0 {
+        storage::set(address_domain, build_index_item_key(table, id), index_len + 1);
+    } else {
+        let data = array![index_len + 1, value].span(); // index and value of the created  entry
+        storage::set_many(address_domain, build_index_item_key(table, id), 0, data, array![250,  252].span());
```

Comment:
Why `array![250, 252]`?

Reply:
All possible values of felt252 had to fit, so 252 bits.
For positions I don't need full range (or reasonably even 64), but I used the greatest number that fits inside 2 felts with value.